faunadb/streams/client.py

```diff
@@ -0,0 +1,115 @@
+        self._data = to_json(expression).encode()
+        try:
+            self.conn = HTTP20Connection(
+                    self._client.domain, port=self._client.port, enable_push=True)
+        except Exception as e:
+            raise StreamError(e)
+
+    def close(self):
+        """
+        Closes the stream subscription by aborting its underlying http request.
+        """
+        if self.conn is None:
+            raise StreamError('Cannot close inactive stream subscription.')
+        self.conn.close()
+        self._state = 'closed'
+
+    def error(self):
+        """
+        Retrieves the error from the event loop in an async implementation.
+        """
+        return self.error
+
+    def subscribe(self, on_event):
+        """Initiates the stream subscription."""
+        if self._state != "idle":
+            raise StreamError('Stream subscription already started.')
+        try:
+            self._state = 'connecting'
+            headers = self._client.session.headers
+            headers["Authorization"] = self._client._auth_header()
+            if self._client._query_timeout_ms is not None:
+                    headers["X-Query-Timeout"] = str(self._client._query_timeout_ms)
+            headers["X-Last-Seen-Txn"] = str(self._client.get_last_txn_time())
+            start_time = time()
+            url_params = ''
+            if isinstance(self._fields, list):
+                url_params= "?%s"%(urlencode({'fields': ",".join(self._fields)}))
+            id = self.conn.request("POST", "/stream%s"%(url_params), body=self._data, headers=headers)
+            self._state = 'open'
+            self._event_loop(id, on_event, start_time)
+        except Exception as e:
+            if callable(on_event):
+                on_event(Error(e), None)
+            raise
+
+    def _event_loop(self, stream_id, on_event, start_time):
+        """ Event loop for the stream. """
+        response = self.conn.get_response(stream_id)
+        if 'x-txn-time' in response.headers:
+            self._client.sync_last_txn_time(int(response.headers['x-txn-time'][0].decode()))
+        try:
+            for push in response.read_chunked():  # all pushes promised before response headers
+                    raw = push.decode()
+                    request_result = self.stream_chunk_to_request_result(response, raw, start_time, time())
+                    event = parse_stream_request_result_or_none(request_result)
+                    if event is not None and hasattr(event, 'txn'):
+                        self._client.sync_last_txn_time(int(event.txn))
+                    on_event(event, request_result)
+                    if self._client.observer is not None:
+                        self._client.observer(request_result)
+        except Exception as e:
+            self.error = e
+            self.close()
+            on_event(Error(e), None)
+
+    def stream_chunk_to_request_result(self, response, raw_chunk, start_time, end_time):
```

Comment:
```suggestion
    def _stream_chunk_to_request_result(self, response, raw_chunk, start_time, end_time):
```

This looks private to me.